internal/apis/config/controller/v1alpha1/defaults.go

```diff
@@ -191,75 +191,91 @@ func SetDefaults_ControllerConfiguration(obj *v1alpha1.ControllerConfiguration)
+	}
+
+	if len(obj.CopiedAnnotationPrefixes) == 0 {
+		obj.CopiedAnnotationPrefixes = defaultCopiedAnnotationPrefixes
 	}
 
-	if obj.LeaderElectionRenewDeadline == time.Duration(0) {
-		obj.LeaderElectionRenewDeadline = defaultLeaderElectionRenewDeadline
+	if obj.NumberOfConcurrentWorkers == nil {
+		obj.NumberOfConcurrentWorkers = &defaultNumberOfConcurrentWorkers
 	}
 
-	if obj.LeaderElectionRetryPeriod == time.Duration(0) {
-		obj.LeaderElectionRetryPeriod = defaultLeaderElectionRetryPeriod
+	if obj.MaxConcurrentChallenges == nil {
+		obj.MaxConcurrentChallenges = &defaultMaxConcurrentChallenges
 	}
 
-	if len(obj.Controllers) == 0 {
-		obj.Controllers = []string{"*"}
+	if obj.MetricsListenAddress == "" {
+		obj.MetricsListenAddress = defaultPrometheusMetricsServerAddress
+	}
+
+	if obj.HealthzListenAddress == "" {
+		obj.HealthzListenAddress = defaultHealthzServerAddress
 	}
 
-	if obj.ACMEHTTP01SolverImage == "" {
-		obj.ACMEHTTP01SolverImage = defaultACMEHTTP01SolverImage
+	if obj.EnablePprof == nil {
+		obj.EnablePprof = &defaultEnableProfiling
 	}
 
-	if obj.ACMEHTTP01SolverResourceRequestCPU == "" {
-		obj.ACMEHTTP01SolverResourceRequestCPU = defaultACMEHTTP01SolverResourceRequestCPU
+	if obj.PprofAddress == "" {
+		obj.PprofAddress = defaultProfilerAddr
 	}
 
-	if obj.ACMEHTTP01SolverResourceRequestMemory == "" {
-		obj.ACMEHTTP01SolverResourceRequestMemory = defaultACMEHTTP01SolverResourceRequestMemory
+	if obj.Logging == nil {
+		obj.Logging = defaultLogging
 	}
+}
 
-	if obj.ACMEHTTP01SolverResourceLimitsCPU == "" {
-		obj.ACMEHTTP01SolverResourceLimitsCPU = defaultACMEHTTP01SolverResourceLimitsCPU
+func SetDefaults_LeaderElectionConfig(obj *v1alpha1.LeaderElectionConfig) {
+	if obj.Enabled == nil {
+		obj.Enabled = &defaultLeaderElect
 	}
 
-	if obj.ACMEHTTP01SolverResourceLimitsMemory == "" {
-		obj.ACMEHTTP01SolverResourceLimitsMemory = defaultACMEHTTP01SolverResourceLimitsMemory
+	if obj.Namespace == "" {
+		obj.Namespace = defaultLeaderElectionNamespace
 	}
 
-	if obj.ACMEHTTP01SolverRunAsNonRoot == nil {
-		obj.ACMEHTTP01SolverRunAsNonRoot = &defaultACMEHTTP01SolverRunAsNonRoot
+	// TODO: Does it make sense to have a duration of 0?
+	if obj.LeaseDuration == time.Duration(0) {
+		obj.LeaseDuration = defaultLeaderElectionLeaseDuration
 	}
 
-	if len(obj.ACMEHTTP01SolverNameservers) == 0 {
-		obj.ACMEHTTP01SolverNameservers = defaultACMEHTTP01SolverNameservers
+	if obj.RenewDeadline == time.Duration(0) {
+		obj.RenewDeadline = defaultLeaderElectionRenewDeadline
 	}
 
-	if obj.ClusterIssuerAmbientCredentials == nil {
-		obj.ClusterIssuerAmbientCredentials = &defaultClusterIssuerAmbientCredentials
+	if obj.RetryPeriod == time.Duration(0) {
+		obj.RetryPeriod = defaultLeaderElectionRetryPeriod
 	}
 
-	if obj.IssuerAmbientCredentials == nil {
-		obj.IssuerAmbientCredentials = &defaultIssuerAmbientCredentials
+	if obj.HealthzTimeout == time.Duration(0) {
+		obj.HealthzTimeout = defaultHealthzLeaderElectionTimeout
```

Comment:
This default was missing.

Reply:
Just curious, how did you find out that the default was missing? Did you notice while looking through the code, or did something break?